logstash-core/lib/logstash/webserver.rb

```diff
@@ -80,6 +83,18 @@ def stop(options={})
 
     private
 
+    def pick_default_port(http_host, range=(9600..9700))
+      range.step(1) do |current_port|
+        begin
+          TCPServer.new(http_host, current_port).close
```

Comment:
I understand your concerns, here, however I don't see a resonable way to add this behaviour inside puma. The only feasible way to make this happen, without any risk of race condition would be to use a system call inside puma, other than that there is no real way to prevent actually to instances to check for the same resource isn't? (actually talking about two codebase copies, isn't?

what do you think?